src/pkg/scan/report/summary.go

```diff
@@ -24,33 +24,67 @@ import (
 	"github.com/pkg/errors"
 )
 
+// CVESet defines the CVE whitelist with a hash set way for easy query.
+type CVESet map[string]struct{}
+
+// Contains checks whether the specified CVE is in the set or not.
+func (cs CVESet) Contains(cve string) bool {
+	_, ok := cs[cve]
+
+	return ok
+}
+
+// Options provides options for getting the report w/ summary.
+type Options struct {
+	// If it is set, the returned summary will not count the CVEs in the list in.
+	CVEWhitelist CVESet
+}
+
+// Option for getting the report w/ summary with func template way.
+type Option func(options *Options)
+
+// WithCVEWhitelist is an option of setting CVE whitelist.
+func WithCVEWhitelist(set *CVESet) Option {
+	return func(options *Options) {
+		options.CVEWhitelist = *set
+	}
+}
+
 // SupportedGenerators declares mappings between mime type and summary generator func.
 var SupportedGenerators = map[string]SummaryGenerator{
 	v1.MimeTypeNativeReport: GenerateNativeSummary,
 }
 
 // GenerateSummary is a helper function to generate report
 // summary based on the given report.
-func GenerateSummary(r *scan.Report) (interface{}, error) {
+func GenerateSummary(r *scan.Report, options ...Option) (interface{}, error) {
 	g, ok := SupportedGenerators[r.MimeType]
 	if !ok {
 		return nil, errors.Errorf("no generator bound with mime type %s", r.MimeType)
 	}
 
-	return g(r)
+	return g(r, options...)
 }
 
 // SummaryGenerator is a func template which used to generated report
 // summary for relevant mime type.
-type SummaryGenerator func(r *scan.Report) (interface{}, error)
+type SummaryGenerator func(r *scan.Report, options ...Option) (interface{}, error)
```

Comment:
This is beyond the scope of this PR but why let this func return a `interface{}` rather than a summary?

Reply:
It's a design consideration. It is a func template for handling different format summary. So, we need a general interface.